### PR TITLE
N21 2075 sync existing course

### PR DESCRIPTION
**Pull request by mrikallab**
# Short Description

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by IgorCapCoder — CHANGES_REQUESTED** [review]

File: src/pages/administration/RoomsOverview.page.vue [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+import { StartExistingCourseSyncDialog } from "@feature-course-sync";
+import { RenderHTML } from "@feature-render-html";
+import { useTitle } from "@vueuse/core";
+import { computed, ComputedRef, onMounted, PropType, ref, Ref } from "vue";
+import { useI18n } from "vue-i18n";
+import { useRoute, useRouter } from "vue-router";
+import { mdiPencilOutline, mdiSync, mdiTrashCanOutline } from "@mdi/js";
+
+type Tab = "current" | "archive";
+// vuetify typing: https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/components/VDataTable/composables/sort.ts#L29-L29
```
Comment (IgorCapCoder) [inline src/pages/administration/RoomsOverview.page.vue:166]: i know this is also in the class overview page code. But why do we need a comment here about vuetify typing?

File: src/pages/administration/RoomsOverview.page.vue [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+	setPagination({ ...pagination.value, limit: itemsPerPage });
+
+	await loadCourseList();
+};
+
+onMounted(() => {
+	onTabsChange(activeTab.value);
+});
+
+const getInstituteTitle: ComputedRef<string> = computed(() => {
```
Comment (IgorCapCoder) [inline src/pages/administration/RoomsOverview.page.vue:365]: this is the 4th page which has the exact same code... isn't it time for a composable? 😅

File: src/pages/administration/RoomsOverview.page.vue [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+			return "Dataport";
+	}
+});
+
+const isCourseSyncDialogOpen: Ref<boolean> = ref(false);
+</script>
+
+<style scoped>
+.v-tabs {
+	margin-bottom: -2px;
```
Comment (IgorCapCoder) [inline src/pages/administration/RoomsOverview.page.vue:383]: isn't it nice to have a page to copy from :D

File: src/pages/administration/RoomsOverview.page.unit.ts [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+			},
+		});
+		return {
+			wrapper,
+			route,
+			router,
+		};
+	};
+
+	const findTableComponen = (wrapper: VueWrapper) => {
```
Comment (IgorCapCoder) [inline src/pages/administration/RoomsOverview.page.unit.ts:92]: i guess you mean ...Component ;)

File: src/pages/administration/RoomsOverview.page.unit.ts [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+				const startSyncButton = wrapper.find(
+					'[data-testid="course-table-start-course-sync-btn"]'
+				);
+
+				expect(startSyncButton.exists()).toBeTruthy();
+				expect(editBtn.exists()).toBeTruthy();
+				expect(deleteBtn.exists()).toBeTruthy();
+			});
+		});
+
```
Comment (IgorCapCoder) [inline src/pages/administration/RoomsOverview.page.unit.ts:353]: As we have a special logic for when to show the sync button, we should test this here.
- course not synced -> show icon (already tested implicitly)
- course already synced -> do not show icon
What happens when course sync feature flag is false?

**Review by IgorCapCoder — CHANGES_REQUESTED** [review]

File: src/modules/feature/course-sync/StartNewCourseSyncDialog.unit.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+			const { wrapper, group } = setup();
+
+			wrapper.getComponent(GroupSelectionDialog).vm.$emit("confirm", group);
+			await nextTick();
+
+			expect(window.location.assign).toHaveBeenCalledWith(
+				`/courses/add?syncedGroupId=${group.id}`
+			);
+		});
+	});
```
[the remark this answers is not in the document]
Comment (mrikallab) [reply src/modules/feature/course-sync/StartNewCourseSyncDialog.unit.ts:76]: Why should we test that no redirect happened, when click on cancel button.. ? 
I think the better thing is to test, if the dialog is closed after emitted cancel event.

File: src/store/env-config-defaults.ts [hunk 7]
```diff
@@ -73,4 +74,5 @@ export const defaultConfigEnvs: ConfigResponse = {
 	FEATURE_SCHULCONNEX_MEDIA_LICENSE_ENABLED: false,
 	FEATURE_AI_TUTOR_ENABLED: false,
 	FEATURE_ROOMS_ENABLED: false,
+	CALENDAR_SERVICE_ENABLED: false,
```
[the remark this answers is not in the document]
Comment (mrikallab) [reply src/store/env-config-defaults.ts:77]: Normally this Feature is not provided in Nuxt or in Server. It is only provided in the old client. I think, if there is a problem with the server and we did not get any response, then this flag should be false.. but this is just my opinion. 
Do you have an Idea when it makes sense to set the default to true? Is "as this flag is true in all instances" an argument to set the default to true ?

File: src/pages/administration/RoomsOverview.page.unit.ts (hunk 5 above)
Comment (mrikallab) [reply src/pages/administration/RoomsOverview.page.unit.ts:353]: It was not possible to test it with exist or visible, but testing if the id is provided on the element works.
I also saw that there is no such feature test in ClassOverview.. but this is not the main thing here..

File: src/pages/administration/RoomsOverview.page.vue (hunk 2 above)
Comment (mrikallab) [reply src/pages/administration/RoomsOverview.page.vue:365]: Sure, maybe for the 5th page 😅.
 I think this PR is big enough to change 4 pages know and test everything. 
 We should take it to the DP refactoring ticket after DP reviewed this beautiful PR.

File: src/serverApi/v3/api.ts [hunk 8]
```diff
@@ -5566,6 +5624,73 @@ export interface OauthConfigResponse {
      */
     jwksEndpoint: string;
 }
+/**
```
[the remark this answers is not in the document]
Comment (mrikallab) [reply src/serverApi/v3/api.ts:5627]: looks for me like someone forgot to generate the api

File: src/serverApi/v3/api.ts [hunk 9]
```diff
@@ -3894,66 +4006,12 @@ export interface LoginRequestBody {
  * @interface LoginResponse
  */
 export interface LoginResponse {
-    /**
```
[the remark this answers is not in the document]
Comment (mrikallab) [reply src/serverApi/v3/api.ts:3897]: export class LoginResponse {
	@ApiProperty()
	accessToken: string;

	constructor(props: LoginResponse) {
		this.accessToken = props.accessToken;
	}
}


that is the login response from the server

**Review by IgorCapCoder — DISMISSED** [review]

**Review by IgorCapCoder — APPROVED** [review]

**Pull request merged**